kerchunk/grib2.py

```diff
@@ -3,7 +3,8 @@
 import io
 import logging
 from collections import defaultdict
-from typing import Iterable, List, Dict, Set
+from typing import Iterable, List, Dict, Set, Optional
+import pandas as pd
```

Comment:
Does kerchunk depend on pandas?
If this is only for the sake of typing, guard with
```
if typing.TYPE_CHECKING:
```

Reply:
This is for the `parse_grib_idx` function to return a Dataframe from parsing the `.idx` file

Reply:
> This is for the parse_grib_idx function to return a Dataframe from parsing the .idx file

The import can be inside the function, so that the module remains importable for those that don't have pandas.

Reply:
> the attrs field is only used from the Dataframe.

It would be good to document and test what attrs should be produced, so that the next stage can run.